Packs/Base/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py

```diff
@@ -0,0 +1,585 @@
+             + use_logo * WEIGHT_HEURISTIC[MODEL_KEY_LOGO_FOUND] * pred_json[MODEL_KEY_LOGO_FOUND]) / total_weight_used
+    return score
+
+
+def get_verdict(pred_json: Dict, is_white_listed: bool) -> Tuple[float, str]:
+    """
+    Return verdict of the url based on the output of the model
+    :param pred_json: output from the model
+    :return:
+    """
+    if is_white_listed:
+        return SCORE_BENIGN, BENIGN_VERDICT
+    score = get_score(pred_json)
+    if pred_json[MODEL_KEY_LOGO_FOUND]:
+        return score, MALICIOUS_VERDICT
+    else:
+        if score < BENIGN_THRESHOLD:
+            return score, BENIGN_VERDICT
+        elif score < SUSPICIOUS_THRESHOLD:
+            return score, SUSPICIOUS_VERDICT
+        else:
+            return score, MALICIOUS_VERDICT
+
+
+def create_dict_context(url, verdict, pred_json, score, is_white_listed, output_rasterize):
+    return {'url': url, 'verdict': verdict, 'pred_json': pred_json, 'score': score, 'is_white_listed': is_white_listed,
+            'output_rasterize': output_rasterize}
+
+
+def extract_created_date(entry_list: List):
+    """
+    Check if domain age is younger than THRESHOLD_NEW_DOMAIN_YEAR year
+    :param entry_list: output of the whois command
+    :return: bool
+    """
+    for entry in entry_list:
+        if is_error(entry):
+            continue
+        else:
+            date_str = entry['EntryContext'].get('Domain(val.Name && val.Name == obj.Name)', {}).get('WHOIS', {}).get(
+                'CreationDate', None)
+            if date_str:
+                date = datetime.strptime(date_str, '%d-%m-%Y')
+                threshold_date = datetime.now() - timedelta(days=THRESHOLD_NEW_DOMAIN_MONTHS * 30)
+                return date > threshold_date
+    return None
+
+
+def get_prediction_single_url(model, url, force_model):
+    is_white_listed = False
+    valid_url, error = is_valid_url(url)
+
+    if not valid_url:
+        return create_dict_context(url, MSG_INVALID_URL + error, {}, SCORE_INVALID_URL, is_white_listed, {})
+
+    # Check domain age from WHOIS command
+    domain = extract_domainv2(url)
+    res = demisto.executeCommand('whois', {'query': domain,
+                                           })
+    is_new_domain = extract_created_date(res)
+
+    # Check is domain in white list -  If yes we don't run the model
+    if in_white_list(model, url):
+        if not force_model:
+            return create_dict_context(url, BENIGN_VERDICT_WHITELIST, {}, SCORE_BENIGN, is_white_listed, {})
+        else:
+            is_white_listed = True
+    # sys.exit(0)
+    # Rasterize html and image
+    res = demisto.executeCommand('rasterize', {'type': 'json',
+                                               'url': url,
+                                               })
+    if len(res) > 0:
+        output_rasterize = res[0]['Contents']
+    else:
+        return_error(MSG_FAILED_RASTERIZE)
+
+    # Create X_pred
+    if isinstance(output_rasterize, str):
```

Comment:
it's better to check the entry type with the function `is_error` and to return the error if it's apply
something like: `if is_error(res): return_error(get_error(res))`

Reply:
sometime it's not error but just a string.